core/src/miner/mem_pool.rs

```diff
@@ -78,6 +79,40 @@ impl ParcelOrigin {
     }
 }
 
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub struct ParcelTimelock {
+    pub block: Option<BlockNumber>,
+    pub timestamp: Option<u64>,
+}
+
+impl PartialOrd for ParcelTimelock {
+    fn partial_cmp(&self, other: &ParcelTimelock) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl Ord for ParcelTimelock {
+    fn cmp(&self, other: &ParcelTimelock) -> Ordering {
+        match (self.block, other.block) {
+            (Some(self_number), Some(other_number)) if self_number != other_number => {
+                return other_number.cmp(&self_number)
+            }
+            (Some(_), _) => return Ordering::Less,
+            (_, Some(_)) => return Ordering::Greater,
+            _ => (),
+        }
+
+        match (self.timestamp, other.timestamp) {
+            (Some(self_timestamp), Some(other_timestamp)) if self_timestamp != other_timestamp => {
+                return other_timestamp.cmp(&self_timestamp)
+            }
+            (Some(_), _) => return Ordering::Less,
```

Comment:
Is `ParcelTimeLock { timestamp: Some(timestamp), .. }` less than `ParcelTimeLock { timestamp: None, .. }`?
What's the definition of the order of the timelock?

Reply:
No, it should be not. I will change it.

A lower value or `None` should be prioritized.
The basic idea is that miners prefer a shorter one so they can earn the fee earlier.